kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/FailoverUrlList.java

```diff
@@ -0,0 +1,61 @@
+package com.provectus.kafka.ui.model;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+public class FailoverUrlList {
+
+  public static final int DEFAULT_RETRY_GRACE_PERIOD_IN_MS = 5000;
+
+  private final Map<Integer, Instant> failures = new HashMap<>();
+  private final List<String> urls;
+  private final int retryGracePeriodInMs;
+
+  private Integer index;
+
+  public FailoverUrlList(List<String> urls) {
+    this(urls, DEFAULT_RETRY_GRACE_PERIOD_IN_MS);
+  }
+
+  public FailoverUrlList(List<String> urls, int retryGracePeriodInMs) {
+    if (urls != null && !urls.isEmpty()) {
+      this.urls = new ArrayList<>(urls);
+      this.index = new Random().nextInt(urls.size());
+    } else {
+      throw new IllegalArgumentException("Expected at least one URL to be passed in constructor");
+    }
+    this.retryGracePeriodInMs = retryGracePeriodInMs;
+  }
+
+  public String current() {
+    return this.urls.get(this.index);
```

Comment:
this is not threadsafe. Even if you updated index within synchronized block there is no guarantee that this change will be visible in other thread (if it is not arranged with [happens-before](https://stackoverflow.com/questions/36383542/understanding-happens-before-and-synchronization/36387640) rule)

Problems here:
1. currentIndex updates in fail() may not be visible when current() will be calles
2. changes in failures map may not be visible in isFailoverAvailable()

How to fix (non-blocking solution):
1.  define `failures = new ConcurrentHashMap<>();` 
2. define `AtomicInteger index = new AtomicInteger(0)`; 
3. fail(...): 
```
public void fail(String url) {
    int currentIndex = this.index.get();
    if ((this.urls.get(currentIndex)).equals(url)) {
      this.failures.put(currentIndex, Instant.now());
      this.index.compareAndSet(currentIndex, (currentIndex + 1) % this.urls.size());
    }
  }
``` 
 in this way you won't need to  use synchronized 

**another** solution is to add `synchronized` to all methods, which may be simpler and ok in this case, since we don't care about performance here much